Spigot-Server-Patches/0028-Lighting-Queue.patch

```diff
@@ -136,41 +136,49 @@ index e8c3dcc24..c47f56ed8 100644
      }
 diff --git a/src/main/java/net/minecraft/server/PaperLightingQueue.java b/src/main/java/net/minecraft/server/PaperLightingQueue.java
 new file mode 100644
-index 000000000..60562f1fd
+index 000000000..1864fa9cc
 --- /dev/null
 +++ b/src/main/java/net/minecraft/server/PaperLightingQueue.java
-@@ -0,0 +1,92 @@
+@@ -0,0 +1,100 @@
 +package net.minecraft.server;
 +
 +import co.aikar.timings.Timing;
 +import it.unimi.dsi.fastutil.objects.ObjectCollection;
 +
 +import java.util.ArrayDeque;
++import java.util.concurrent.Executors;
++import java.util.concurrent.ForkJoinPool;
++import java.util.concurrent.TimeUnit;
 +
 +class PaperLightingQueue {
 +    private static final long MAX_TIME = (long) (1000000000 / 20 * .95);
 +    private static int updatesThisTick;
 +
++    private static ForkJoinPool pool = (ForkJoinPool) Executors.newWorkStealingPool();
 +
 +    static void processQueue(long curTime) {
 +        updatesThisTick = 0;
 +
 +        final long startTime = System.nanoTime();
 +        final long maxTickTime = MAX_TIME - (startTime - curTime);
 +
++
 +        START:
 +        for (World world : MinecraftServer.getServer().getWorlds()) {
 +            if (!world.paperConfig.queueLightUpdates) {
 +                continue;
 +            }
 +
 +            ObjectCollection<Chunk> loadedChunks = ((WorldServer) world).getChunkProviderServer().chunks.values();
-+            for (Chunk chunk : loadedChunks.toArray(new Chunk[loadedChunks.size()])) {
```

Comment:
idk why this was there so i removed it

Reply:
again..... if you don't understand why this is happening....

Reply:
you dont need to specify the size in the type reference array yknow